include/service/entities/plugin_service.js

```diff
@@ -161,7 +185,35 @@ module.exports = function PluginServiceModule(pb) {
      * initialized successfully within this instance.
      */
     PluginService.prototype.getActivePluginNames = function() {
-        return Object.keys(ACTIVE_PLUGINS);
+        var globalPlugins = [];
+        if(ACTIVE_PLUGINS[GLOBAL_SITE]) {
+            globalPlugins = Object.keys(ACTIVE_PLUGINS[GLOBAL_SITE]);
+        }
+        var sitePlugins = [];
+        if(ACTIVE_PLUGINS[this.site]) {
+            sitePlugins = Object.keys(ACTIVE_PLUGINS[this.site]);
+        }
+        var merged = util.dedupeArray(sitePlugins.concat(globalPlugins));
+        return merged;
+    };
+
+    PluginService.prototype.getAllActivePluginNames = function() {
+        var pluginNames = [];
+        var siteNames = Object.keys(ACTIVE_PLUGINS);
+        for( var i = 0; i < siteNames.length; i++ ) {
+            var sitePluginNames = Object.keys(ACTIVE_PLUGINS[siteNames[i]]);
+            for ( var j = 0; j < sitePluginNames.length; j++ ) {
+                pluginNames.push(siteNames[i] + '_' + sitePluginNames[j]);
+            }
+        }
+    };
```

Comment:
Why not just return a hash where the key is the site name and the value is an array of active plugin UIDs.